environment/template.go

```diff
@@ -85,29 +85,31 @@ type Template struct {
 	Filename      string
 	DefaultParams map[string]string
 	Content       string
+	Version       string
 }
 
 var (
 	specialCharRegexp = regexp.MustCompile("[^a-z0-9]")
 	variableRegexp    = regexp.MustCompile(`\${([A-Z_0-9]+)}`)
 )
 
-func newTemplate(filename string, defaultParams map[string]string) *Template {
+func newTemplate(filename string, defaultParams map[string]string, version string) *Template {
```

Comment:
I use the pointer later on - when the content of the template is set. And it is easier to create the pointer directly in the constructor then in a loop by creating a copy of the value for the whole array. And since there is only one state of each template per one request (see that the instances of templates are [created](https://github.com/fabric8-services/fabric8-tenant/pull/680/files#diff-548bfeebae84ae77e57356c9ba042b4aR40) on demand - by function and not as variable), then I don't see any point in removing the pointer in the constructor

Reply:
it's just that in general, you should only return a pointer if the element needs to be changed by the caller but the variable belongs to the callee. Same with func arguments: only use a pointer if fields needs to be changed, otherwise, pass a copy.

Reply:
I think that the topic of pointers/values is more complicated. But this is a different story for sure. 
I can do this 6fc3ff6 if you think that it looks better/is cleaner

Reply:
yes, but that's a rule that I try to use when I need to decide whether to use pointers or not. Also, golang is very fast at copying memory blocks, so passing a copy isn't so expensive.
I'd rather not use pointers when really needed, actualy

Reply:
Anyway, this is exactly the case when I need to change the instance of the template - to set the content and potentially change the version. So, are you fine with the proposed solution?